posts/navarroaxel/17.git-rebase-interactive.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Rewriting the history with Git rebase
+published: true
+description: How to squash commits and keep your branch up-to-date using git rebase 
+tags: 'git, bash, productivity, tooling'
+cover_image: ./assets/git-rebase.jpg
+id: 1949841
+---
+
+As when we work in a new feature or bug fix it's recommended to keep our branch up-to-date with the target branch (`main`, `development`, etc.), we can use the [`git rebase`](https://git-scm.com/docs/git-rebase) command instead of `git merge` in order to keep our Git history cleaner.
+
+The `git rebase` command reapplies our commits on top of another branch, rewriting the Git history; this is not recommended on shared branches (like `main`) because it causes divergence (and therefore conflict solving) and panic on your teammates 😬 since it is sometimes hard to resolve.
+
+## The basics
+
+To be clear about how `git rebase` works we're going to use the simplest example: you have 3 commits in your feature branch and you want to get the latest changes from `main`, so you can simply run the following command:
+
+```bash
+git rebase main
+```
+
+If you're lucky to get no conflicts then you'll have a new Git history with your 3 commits on the top of the `main` branch and you just need to override (`--force`) the remote branch with your new history.
+
+```bash
+git push --force origin feature/abc
```

Comment:
Why not `--force-with-lease`?

Reply:
not a wide knowledge about ` --force-with-lease`  and I don' t want to talk about it specifically.